### PR TITLE
[EBPF] event_monitor interfaces refactor

### PR DESCRIPTION
**Pull request by val06**
### What does this PR do?

small refactor in event_monitoring interfaces for clarity

### Motivation

- get more clear interfaces and avoid confusions
- follow idiomatic go for naming conventions

### CODE REVIEW

**Review by amitslavin — APPROVED** [review]

File: pkg/eventmonitor/consumer.go [hunk 1]
```diff
@@ -8,15 +8,15 @@ package eventmonitor
 
 import "github.com/DataDog/datadog-agent/pkg/security/probe"
 
-// EventConsumer event consumer
-type EventConsumer interface {
-	probe.EventConsumerInterface
+// EventConsumerHandler provides an interface for event consumer handlers
```
Comment (gjulianm) [inline pkg/eventmonitor/consumer.go:11]: I would add to these comments the relationship that these two interfaces have with one another (AFAIK, the `EventConsumer` starts/stops the `EventConsumerHandler`).

**Review by gjulianm — APPROVED** [review]
LGTM, just one minor comment.

**Review by safchain — APPROVED** [review]

**Review by kkhor-datadog — APPROVED** [review]

**Review by hmahmood — APPROVED** [review]

File: pkg/eventmonitor/examples/consumer.go [hunk 2]
```diff
@@ -5,6 +5,7 @@
 
 //go:generate go run github.com/DataDog/datadog-agent/pkg/security/generators/event_copy -scope "(fc *SimpleEventConsumer)" -pkg examples -output ./event_copy.go SimpleEvent .
 
+// Package examples package provides an example of how to use the event monitor module using a basic consumer
```
Comment (guyarb) [inline pkg/eventmonitor/examples/consumer.go:8]:
```suggestion
// Package examples provides an example of how to use the event monitor module using a basic consumer
```

File: pkg/security/probe/probe.go [hunk 3]
```diff
@@ -65,18 +65,24 @@ type EventHandler interface {
...
 	Copy(_ *model.Event) any
 	EventTypes() []model.EventType
 }
 
+// IDer provides unique ID for each event consumer
+type IDer interface {
+	// ID returns the ID of the event consumer
+	ID() string
+}
+
```
Comment (guyarb) [inline pkg/security/probe/probe.go:82]: why did you declare twice on the same interface?

Comment (val06) [reply pkg/security/probe/probe.go:82]: moved all event consumer interfaces to a platform-agnostic file (without go:build tags)

**Pull request merged**